Main.java

```diff
@@ -0,0 +1,156 @@
+            return -1;
+        }
+        return Binary2Decimal(BinaryString.substring(1), value);
+    }
+
+    public static long NthFibonacci(int position) {
+        if (position == 0) {
+            return 0;
+        }
+        if (position == 1) {
+            return 1;
+        }
+        return NthFibonacci(position - 1) + NthFibonacci(position - 2);
+    }
+
+    public static long CountConsonants(String InputString) {
+        return countConsonantsHelper(InputString, 0);
+    }
+
+    private static int countConsonantsHelper(String InputString, int index) {
+     if (index == InputString.length()) {
+      return 0;
+    }
+     char character = InputString.charAt(index);
+     boolean isConsonant = (character >= 'a' && character <= 'z' || character >= 'A' && character <= 'Z') && 
+                        character != 'a' && character != 'e' && character != 'i' && character != 'o' && character != 'u' &&
+                        character != 'A' && character != 'E' && character != 'I' && character != 'O' && character != 'U';
+     return (isConsonant ? 1 : 0) + countConsonantsHelper(InputString, index + 1);
+    }
+
+    private static String convertToCamelCase(String InputString, int index, String result, boolean capitalize) {
+        if (index == InputString.length()) return result;
+        char character = InputString.charAt(index);
+        if ((character >= 'A' && character <= 'Z') || (character >= 'a' && character <= 'z') || character == ' ' || character == '_') {
+            character = (character >= 'A' && character <= 'Z') ? (char) (character + 32) : character; 
+            if (character == ' ' || character == '_') {
+                return convertToCamelCase(InputString, index + 1, result, true);
+            }
+            result += capitalize ? (char) (character - 32) : character; 
+            return convertToCamelCase(InputString, index + 1, result, false);
+        }
+        return convertToCamelCase(InputString, index + 1, result, capitalize); 
+    }
+
+    public static int countPalindrome(String InputString , int count) {
+        if(InputString.length() == 1){
+            return count + 1;
+        }
+        if (InputString.isBlank()) {
+            return count ;
+        }
+        if (isValidPalindrome(InputString)){
+                count++;
+            }
+        if(InputString.charAt(0) == InputString.charAt(InputString.length()-1)){
+                count++;
+        }
+        else{
+                count = count + 2;
+            }
+        return countPalindrome(InputString.substring(1,InputString.length()-1),count);
+        }
+        public static boolean isValidPalindrome(String s){
+
+        if(s.isEmpty() || s.length() == 1){
+            return true;
+        }
+            if(s.charAt(0) == s.charAt(s.length()-1)){
+                return isValidPalindrome(s.substring(1,s.length()-1));
+            }
+            else{
+                return false;
+            }
+    }
+}
+
+

```

Comment:
avoid unnecessary spaces in your programs as it leads to increment in lines of code